recipes/karchive/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+from conan.tools.files import rename
+from conans import ConanFile, CMake, tools
+import shutil
+
+
+class KArchiveConan(ConanFile):
+    name = "karchive"
+    license = ("BSD-2-Clause", "LGPL-2.0-only", "LGPL-2.0-or-later", "LGPL-3.0-only", "LicenseRef-KDE-Accepted-LGPL")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://api.kde.org/frameworks/karchive/html/index.html"
+    topics = ("archive", "compression")
+    description = "KArchive provides classes for easy reading, creation and manipulation of archive formats like ZIP and TAR."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "with_lzma": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_lzma": True
+    }
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package_multi"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("extra-cmake-modules/5.75.0")
+
+    def requirements(self):
+        self.requires("qt/5.15.2")
+        self.requires("zlib/1.2.11")
+        self.requires("bzip2/1.0.8")
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        rename(self, "karchive-{}".format(self.version), self._source_subfolder)
```

Comment:
We can do that smarter by now:

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

Reply:
that is a good one ;)